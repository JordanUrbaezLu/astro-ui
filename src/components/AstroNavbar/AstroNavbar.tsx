import * as React from "react";
import styles from "./AstroNavbar.module.scss";

export interface AstroNavbarLogoProps {
  /**
   * The link for the AstroNavbar's logo
   */
  link?: string;

  /**
   * The url for the AstroNavbar's logo
   */
  url?: string;
}

export interface AstroNavbarProps {
  /**
   * The content for the AstroNavbar
   */
  children?: React.ReactNode;
  /**
   * The props for the AstroNavbar's logo
   */
  logoProps?: AstroNavbarLogoProps;
}

const AstroNavbar: React.FC<AstroNavbarProps> = ({ children, logoProps }) => {
  return (
    <nav className={styles.astroNavbarContainer}>
      <a
        className={styles.logoContainer}
        href={logoProps?.link ? logoProps.link : "#"}
      >
        <img src={logoProps?.url} />
      </a>
      <ul className={styles.listContainer}>{children}</ul>
    </nav>
  );
};

export default AstroNavbar;
