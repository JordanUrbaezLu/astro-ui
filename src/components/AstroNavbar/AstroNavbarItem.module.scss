@use "../../utility/globals";

.astroNavbarItemLinkContainer {
  color: black;
  text-decoration: none;
  padding: 20px;
}

.primary {
    &:hover {
        background-color: globals.$primary-color;
      }
}

.secondary {
  &:hover {
    background-color: globals.$secondary-color;
  }
}
