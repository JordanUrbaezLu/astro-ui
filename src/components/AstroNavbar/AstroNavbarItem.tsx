import * as React from "react";
import styles from "./AstroNavbarItem.module.scss";

export interface AstroNavbarItemProps {
  /**
   * The content for the AstroNavbarItem
   */
  children: React.ReactNode;
  /**
   * The url link for the AstroNavbarItem
   */
  link: string;
}

const AstroNavbarItem: React.FC<AstroNavbarItemProps> = ({
  children,
  link,
}) => {
  return (
    <li className={styles.astroNavbarItemContainer}>
      <a href={link}>{children}</a>
    </li>
  );
};
export default AstroNavbarItem;
