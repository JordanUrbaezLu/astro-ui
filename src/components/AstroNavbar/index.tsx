export * from "./AstroNavbar";
export * from "./AstroNavbarItem";
