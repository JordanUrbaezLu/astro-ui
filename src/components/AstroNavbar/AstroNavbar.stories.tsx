import AstroNavbar from "./AstroNavbar";
import AstroNavbarItem from "./AstroNavbarItem";

export default {
  component: AstroNavbar,
  title: "Components/AstroNavbar",
};

export const Default = () => {
  return (
    <AstroNavbar>
      <AstroNavbarItem link="https://github.com/JordanUrbaezLu/astro-ui-library">
        Github
      </AstroNavbarItem>
      <AstroNavbarItem link="https://github.com/JordanUrbaezLu/astro-ui-library">
        Github
      </AstroNavbarItem>
      <AstroNavbarItem link="https://github.com/JordanUrbaezLu/astro-ui-library">
        Github
      </AstroNavbarItem>
    </AstroNavbar>
  );
};

Default.parameters = {
  layout: "fullscreen",
};

export const Logo = () => {
  return (
    <AstroNavbar
      logoProps={{
        link: "https://github.com/JordanUrbaezLu/astro-ui-library",
        url: "https://www.pngkey.com/png/full/141-1415366_arctic-cat-logo-github-png.png",
      }}
    >
      <AstroNavbarItem link="https://github.com/JordanUrbaezLu/astro-ui-library">
        Github
      </AstroNavbarItem>
      <AstroNavbarItem link="https://github.com/JordanUrbaezLu/astro-ui-library">
        Github
      </AstroNavbarItem>
      <AstroNavbarItem link="https://github.com/JordanUrbaezLu/astro-ui-library">
        Github
      </AstroNavbarItem>
    </AstroNavbar>
  );
};

Logo.parameters = {
  layout: "fullscreen",
};

export const Secondary = () => {
    return (
      <AstroNavbar
        logoProps={{
          link: "https://github.com/JordanUrbaezLu/astro-ui-library",
          url: "https://www.pngkey.com/png/full/141-1415366_arctic-cat-logo-github-png.png",
        }}
        variant="secondary"
      >
        <AstroNavbarItem link="https://github.com/JordanUrbaezLu/astro-ui-library">
          Github
        </AstroNavbarItem>
        <AstroNavbarItem link="https://github.com/JordanUrbaezLu/astro-ui-library">
          Github
        </AstroNavbarItem>
        <AstroNavbarItem link="https://github.com/JordanUrbaezLu/astro-ui-library">
          Github
        </AstroNavbarItem>
      </AstroNavbar>
    );
  };
  
  Secondary.parameters = {
    layout: "fullscreen",
  };
