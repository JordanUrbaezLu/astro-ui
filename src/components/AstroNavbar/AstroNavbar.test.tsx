import { render } from "@testing-library/react";
import AstroNavbar from "./AstroNavbar";

describe("AstroCalendar", () => {
  test("Should render component correctly.", () => {
    const { container } = render(<AstroNavbar />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
