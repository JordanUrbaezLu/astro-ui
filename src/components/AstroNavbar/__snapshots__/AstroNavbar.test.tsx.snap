// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AstroCalendar Should render component correctly. 1`] = `
<nav
  class="astroNavbarContainer"
>
  <a
    class="logoContainer"
    href="#"
  >
    <img />
  </a>
  <ul
    class="listContainer"
  />
</nav>
`;
